### ISSUE TITLE
`assembly ("memory-safe")` not supported

### ISSUE THREAD

**Issue by PaulRBerg** [protofire/solhint#398]
Solidity v0.8.13 [introduced](https://docs.soliditylang.org/en/v0.8.13/assembly.html) a dialect that lets users mark a piece of assembly code as memory safe:

```solidity
assembly ("memory-safe") {
    ...
}
```

This dialect is not currently supported by Solhint, at least not in the visual interface in VSCode:

<img width="752" alt="Screenshot 2023-02-05 at 8 03 54 PM" src="https://user-images.githubusercontent.com/8782666/216836852-363572c2-26e3-4c4f-a0e2-16c612db3b65.png">

It's odd, however, because when running `solhint` as a CLI, no error is thrown.

**Comment by fvictorio** [comment]
Hey @PaulRBerg, I guess you are using [Juan Blanco's solidity extension](https://github.com/juanfranblanco/vscode-solidity)? If that's the case, i think the error should be reported there if the CLI works fine.

**Comment by PaulRBerg** [comment]
Yes, I am using Juan's extension, and I think you are right. I will close this issue here and re-open it there. Sorry.
